src/main/java/net/syscon/elite/service/BookingService.java

```diff
@@ -626,13 +626,17 @@ public MilitaryRecords getMilitaryRecords(final Long bookingId) {
     }
 
     @VerifyBookingAccess
-    public List<CourtCase> getOffenderCourtCases(final Long bookingId) {
+    public List<CourtCase> getOffenderCourtCases(final Long bookingId, final boolean activeOnly) {
         return offenderBookingRepository.findById(bookingId)
-                .map(OffenderBooking::getCourtCases)
+                .map(booking -> activeOnly ? active(booking.getCourtCases()) : booking.getCourtCases())
                 .map(CourtCaseTransformer::transform)
                 .orElseThrow(EntityNotFoundException.withId(bookingId));
     }
 
+    private Collection<OffenderCourtCase> active(final Collection<OffenderCourtCase> courtCases) {
+        return courtCases.stream().filter(OffenderCourtCase::isActive).collect(Collectors.toUnmodifiableList());
```

Comment:
Maybe put the filter in the main stream in the calling method?  Less complex and no need for a second stream on the same thing...

Reply:
After discussion I see where you are coming from... ignore me :)

Reply:
I will push this down in to the Domain ;)